### PR TITLE
Ss2 build reference

### PR DESCRIPTION
**Pull request by jishuxu**
main changes.
1. add demo json. single chrom chr21 and 10k fastq PE reads(extracted from aligned bam file)
2. create "BuildRef bundle" workflow. Generate index files for STAR and rsem.
3. Some changes in pipeline wdl. 
a) change reference package to use tar file instead of tar.gz
b) increase featureCounts task memory to 15G
c) switch to standard hg19 reference bundle. Previous KCO bundle generated warning in FeatureCounts task.
d) add a workflow configuration json which can turn off CallCaching

### CODE REVIEW

**Comment by kbergin** [comment]
Can you add a WDL that generates that downsampled data in `smartseq2_single_sample/Demo/ss2_single_sample_demo.json` ?

**Comment by TimothyTickle** [comment]
I think that the submission/ingest process makes all json files but the analysis.json but that one happens after analysis. Not sure reference bundle scripts would make those.

File: smartseq2_single_sample/ss2_single_sample.wdl [hunk 1]
```diff
@@ -42,7 +42,7 @@ task FeatureCountsUniqueMapping {
   }
   runtime {
     docker:"humancellatlas/star_dev:v1"
```
Comment (ambrosejcarr) [inline smartseq2_single_sample/ss2_single_sample.wdl]: Do we have a standard tagging method for docker images? I'm looking and I see a mix of:
1. latest
2. v.major
3. v.major.minor x2

File: smartseq2_single_sample/BuildRef/BuildReferenceBundle.wdl [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+workflow BuildRef {
+  File fasta
+  File gtf
+  
+  call BuildStarReference {
+    input:
+      ref_fasta = fasta,
+      gtf_file = gtf
+  }
+  call BuildRsemReference{
```
Comment (ambrosejcarr) [inline smartseq2_single_sample/BuildRef/BuildReferenceBundle.wdl]: Inconsistency in formatting: below the inputs are input=value, above they are input = value.

Comment (dshiga) [reply smartseq2_single_sample/BuildRef/BuildReferenceBundle.wdl]: I find the extra spaces a bit clearer, input = value. We need that linter.

**Comment by kbergin** [comment]
I wanted to add a comment that I know I didn't clearly state having a WDL to create the testing data in the ticket or AC for the ticket for this work. I think we discussed it a little in a meeting, but I may be totally making that up. If you don't think you have the capacity this sprint to add in the WDL we can bump it into a new ticket! I'm sorry about that, I will work on being more detailed and clear in tickets and ACs.

**Comment by kbergin** [comment]
@TimothyTickle Sorry my comment wasn't clear. I didn't mean that the WDL would generate the json file, I meant it should generate the actual data. Whatever steps were taken to downsample the fasta files in WDL form so that the output of the WDL is a smaller fasta file (in this case, the demo data).

**Comment by ambrosejcarr** [comment]
Kylee, I think the tool I used for 10x could be adapted to work on this
data if I steal Jishu's STAR task. I'll talk to her on Monday about it.
Shouldn't take more than an hour to create and test if we want to go that
route.

**Comment by jishuxu** [comment]
Split wdl to build reference for RSEM and STAR separately and some minor formatting changes

**Comment by jishuxu** [comment]
I have a script in wdl to do down-sample and used to generated demo data. I can talk with Ambrose and see which one we want add into PR.

**Pull request merged**